test/gateway/staker_proxy/transfer_token.js

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+'use strict';
+
+const MockToken = artifacts.require('./MockToken.sol');
+const StakerProxy = artifacts.require('./StakerProxy.sol');
+const Utils = require('../../test_lib/utils');
+
+contract('StakerProxy.transferToken()', (accounts) => {
+  const [deployer, owner, someoneElse] = accounts;
+  const transferValue = '1000';
+
+  let stakerProxy;
+
+  beforeEach(async () => {
+    // Account deployer works, as no composer is queried for `transferToken`.
+    stakerProxy = await StakerProxy.new(
+      owner,
+      { from: deployer },
+    );
+  });
+
+  it('should send the tokens to the desired recipient', async () => {
+    const mockToken = await MockToken.new({ from: deployer });
+    await mockToken.transfer(
+      stakerProxy.address,
+      transferValue,
+      { from: deployer },
+    );
+
+    await stakerProxy.transferToken(
+      mockToken.address,
+      someoneElse,
+      transferValue,
+      { from: owner },
+    );
+
+    const balance = await mockToken.balanceOf.call(someoneElse);
+
+    assert.strictEqual(
+      balance.toString(10),
+      transferValue,
+      'The balance of `someoneElse` should equal the transfer value after the transfer.',
+    );
+  });
+
+  it('should fail if not called by the owner', async () => {
+    // Other function arguments are only placeholders as the call will fail on the modifier.
+    await Utils.expectRevert(
+      stakerProxy.transferToken(
+        someoneElse,
+        someoneElse,
+        transferValue,
+        { from: someoneElse },
+      ),
+      'This function can only be called by the owner.',
+    );
+  });
+
```

Comment:
Negative unit case can be added for use case "when owner doesn't have sufficient balance and he is doing transferToken".

Reply:
In this case the unit under test is the `StakerProxy`. It is not in the scope of the `StakerProxy` to test the token that is used for the transfer (and should fail on insufficient funds).

What do you think? Should I still add a test? I wonder which revert message is returned, the one from the token or the one from the surrounding `StakerProxy` 🤔

Reply:
Error will be thrown from token transfer method because of SafeMath -ve calculation.
https://github.com/OpenST/mosaic-contracts/blob/develop/contracts/gateway/EIP20Token.sol#L257
I think we will get "VM exception" error 🤔 .

No need to add this test as it's out of scope of StakerProxy.